presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/ShardCleaner.java

```diff
@@ -327,35 +332,79 @@ synchronized void cleanLocalShards()
     @VisibleForTesting
     void cleanBackupShards()
     {
+        Set<UUID> processing = newConcurrentHashSet();
+        BlockingQueue<UUID> completed = new LinkedBlockingQueue<>();
+        long start = System.nanoTime();
+        boolean fill = true;
+
         while (!Thread.currentThread().isInterrupted()) {
-            List<UUID> uuids = dao.getCleanableShardsBatch(maxTimestamp(backupCleanTime));
-            if (uuids.isEmpty()) {
+            // get a new batch if any completed and we are under the batch size
+            Set<UUID> uuids = ImmutableSet.of();
+            if (fill && (processing.size() < 1000)) {
```

Comment:
This needs to match the LIMIT value in the SQL query. I could make that a constant with string concatenation, but that seems worse.